zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/AQLQueryParser.scala

```diff
@@ -232,8 +248,44 @@ class AQLQueryParser extends IQueryParser {
     (prefix, aql)
   }
 
+  private def parseGlobalAggr(globalAggr: GlobalAggregateStatement,
+                          varMap: Map[String, AQLVar],
+                          sourceVar: String = "$g"
+                         ): (String, String) = {
+
+    val (forPrefix, forWrap) = (s"for $sourceVar in (", ")")
+    val producedVar = mutable.Map.newBuilder[String, AQLVar]
+    val aggr = globalAggr.aggregate
+    val (aggrNameMap, returnVar) =
+      varMap.get(aggr.fieldName) match {
+        case Some(aqlVar) =>
+          val (dataType, aqlAggExpr, returnVar) = AQLFuncVisitor.translateGlobalAggr(aqlVar.field, aggr.func, sourceVar)
+          producedVar += aggr.as -> AQLVar(new Field(aggr.as, dataType), s"$sourceVar.${aggr.as}")
```

Comment:
this is not true for the global aggregation case which only returns a scalar value not a record. thus, it can't be access by path expression.